HighLight and Jump to Missing Messages

Renu-Priya411 opened the pull request.

Renu-Priya411 committed:
DltCounter plugin added to find the missing counter messages

Changes made for displaying only the missing messages
Signed-off by : RenuPriya Krishnamoorthy <renupriya.ka.krishnamoorthy@partner.bmwgroup.com>

Renu-Priya411 committed:
DltCounter plugin added to find the missing counter messages

Changes made for displaying only the missing messages
Signed-off by : RenuPriya Krishnamoorthy <renupriya.ka.krishnamoorthy@partner.bmwgroup.com>

Renu-Priya411 committed:
Changes in mainwindow, qdltcontrol and dltCounter plugin

Export feature for dltCounter Plugin has been implemented
Signed-off by : RenuPriya Krishnamoorthy <renupriya.ka.krishnamoorthy@partner.bmwgroup.com>

Renu-Priya411 committed:
Changes in mainwindow, qdltcontrol and dltCounter plugin

Export feature for dltCounter Plugin has been implemented
Signed-off by : RenuPriya Krishnamoorthy <renupriya.ka.krishnamoorthy@partner.bmwgroup.com>

Renu-Priya411 committed:
Dlt Counter Plugin : Used for finding the missing sequence of messages using the counter and the context ID values. The missing sequence of numbers will be visible in a new window. The data can be exported to local machine as well.

Changes made in mainwindow.cpp, qdltcontrol and dltCounterPlugin

Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@partner.bmwgroup.com>

Renu-Priya411 committed:
Dlt Counter Plugin : Used for finding the missing sequence of messages using the counter and the context ID values. The missing sequence of numbers will be visible in a new window. The data can be exported to local machine as well.

Changes made in mainwindow.cpp, qdltcontrol and dltCounterPlugin

Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@partner.bmwgroup.com>

hannesa2 commented:
Btw, the commits are a mess. Please can you rebase to master ?

```
git fetch upstream
git rebase upstream 
```

alexmucde commented:
@Renu-Priya411 Please move both functions from Mainwindow into plugin. Plugin gets access to all DLTMsgs in initMsg function, when DLt file is loaded. There you can create the Map regarding the mcnt values.

Renu-Priya411 committed:
Changes in dltcounterplugin.cpp

The data is completely handled within the DLTCounterPlugin. Changes in mainwindow.cpp and qdltcontrol.cpp has been removed.
Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@bti.bmwgroup.com>

alexmucde commented on src/mainwindow.cpp:
```diff
@@ -2106,10 +2115,13 @@ void MainWindow::reloadLogFileFinishFilter()
             }
         }
-
+        project.ecu->clear();
```
Check if this line was removed in master. Remove also if needed.

alexmucde commented on src/mainwindow.cpp:
```diff
@@ -51,6 +51,15 @@
+ * for MSVC compilers.
+ **/
+extern "C" {
+    #include "dlt_user.h"
```
Remove if not needed

alexmucde commented:
Please rebase to master and only Plugin should appear in the changed files.

Renu-Priya411 committed:
Conflicts Resolved

Conflicts in mainwindow.cpp has been resolved
Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@bti.bmwgroup.com>

Renu-Priya411 committed:
Conflicts Resolved

Resolved the conflicts in mainwindow.cpp

Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@bti.bmwgroup.com>

Renu-Priya411 committed:
Spacing Issue Resolved in mainwindow.h

Changes in mainwindow.h

Signed-off by : Renu Priya Krishnamoorthy <RenuPriya.KA.Krishnamoorthy@bti.bmwgroup.com>

alexmucde committed:
Update form.cpp

Renu-Priya411 committed:
Changes in DLTCounterPlugin.cpp

The fileSaveDialog has been implemented. Jumping to previous or next message from the missing messages has been implemented.
Conflicts has been fixed

Signed-off by : Renu Priya Krishnamoorthy
<RenuPriya.KA.Krishnamoorthy@bti.bmwgroup.com>
